environment.rb

```diff
@@ -0,0 +1,98 @@
+require 'securerandom' #Gem to generate ids
+
+module Environment
+  class Depot
+      attr_accessor :packs
+
+      def initialize
+        @packs = {
+          simple_transportation_pack: { 
+              intelligence: %i[:cellphone],
+              items: %i[medipack chevy_versa] 
+          },
+          standard_transportation_pack: { 
+              intelligence: %i[cellphone antenna], 
+              arsenal: %i[:colt_1911],
+              items: %i[handcuffs medipack chemestry chevy_versa] 
+          },
+          simple_mission_pack: { 
+              intelligence: %i[infopack laptop cellphone antenna], 
+              arsenal: %i[:colt_1911],
+              items: %i[handcuffs medipack chemistry financial] 
+          },
+          standard_mission_pack: { 
+              intelligence: %i[infopack laptop cellphone antenna], 
+              arsenal: %i[remington_870 colt_1911 machete hatchet],
+              items: %i[handcuffs medipack chemistry financial] 
+          }
+        }
+      end
+  end
+
+  class Control
+      attr_accessor :missions
+
+      states = %i[paused aborted accomplished failed]
+
+      states.each do |state|
+        define_method("set_mission_to_#{state}") do |mission|
+          @missions[mission][:status] = state
+        end
+      end
+
+      def initialize
+        @missions = {}
+      end
+      
+      def new_mission(name:, objective:, pack:)
```

Comment:
In some cases, the use of parenthesis is not recommended, in Ruby, methods without them are preferred.